views/postgres/upgrade.erb

```diff
@@ -0,0 +1,77 @@
+<% upgrade_stage_descriptions = {
+    "start" => "Start",
+    "wait_for_standby" => "Wait for standby",
+    "wait_for_maintenance_window" => "Wait for maintenance window",
+    "wait_fence_primary" => "Wait to fence primary",
+    "upgrade_standby" => "Upgrade standby",
+    "upgrade_failed" => "Upgrade failed",
+    "update_metadata" => "Update metadata",
+    "wait_takeover" => "Wait for takeover",
+    "prune_old_servers" => "Prune old servers"
+} %>
+
+<div class="p-6">
+  <h3 class="text-2xl font-bold leading-7 text-gray-900 sm:truncate sm:text-2xl sm:tracking-tight">Upgrade</h3>
+
+  <% if @pg.display_state == "creating" %>
+    <p class="text-gray-500 mt-2">Upgrade operation is not available while the database is being created.</p>
+  <% elsif @pg.needs_upgrade? %>
+    <% if @pg.upgrade_stage == "upgrade_failed" %>
+      <p>Database upgrade failed. We have reverted the database to the previous version, and it is now available again.</p>
+    <% else %>
+      <p>Database upgrade is in progress.</p>
+    <% end %>
+    <% database_available_upgrade_stages = ["start", "wait_for_standby", "wait_for_maintenance_window"]
+    
+    upgrade_status = @pg.upgrade_progress.map do |stage|
+      [
```

Comment:
Indentation starting here looks off. Can we keep a 2-space indentation level here?